app/shell-window/ui/tabs.js

```diff
@@ -156,8 +156,47 @@ function onUpdateTab (page) {
   getTabEl(page, tabEl => yo.update(tabEl, drawTab(page)))
 }
 
+/**
+ * Check if the page url has changed to a different service or public ID to avoid triggering freeing of handles
+ * for webapps (using pushstate/or hash url changes).
+ *
+ * This assumes that no one would be having a different webapp at the same service/id on a different path. We need
+ * to assume this to enable pushstate.
+ * @param  {String} url  string of the new url
+ * @param  {String} prevUrl string of page's previous url
+ * @return {Bool}      has the page changed for SAFE's concerns / to free the app handles automatically?
+ */
+function pageURLChangedForSafe( url, prevUrl )
+{
+  if( !url || !prevUrl )
+  {
+    return false
+  }
+
+  let newUrlObj = urlParse( url )
+  let prevUrlObj = urlParse( prevUrl )
+
+  if( newUrlObj.protocol.startsWith( 'safe' ) &&  newUrlObj.hostname !== prevUrlObj.hostname )
```

Comment:
Can also be refactored as 
`return newUrlObj.protocol.startsWith( 'safe' ) &&  newUrlObj.hostname !== prevUrlObj.hostname`.

We can avoid the if else if possible

Reply:
>Is there any specific reason why we are checking for startsWith('safe') instead of newUrlObject.protocol === 'safe:'

figured we might want `safe-auth` to be caught here too.

Reply:
I think, it would be better to mention these two protocols than being generic with `safe*`. 
Eg, `newUrlObject.protocol === 'safe:' && newUrlObject.protocol === 'safe-auth:'`